src/main/scala/chisel3/util/util.scala

```diff
@@ -11,4 +11,15 @@ package object util {
   type ValidIO[+T <: Data] = chisel3.util.Valid[T]
   val ValidIO = chisel3.util.Valid
   val DecoupledIO = chisel3.util.Decoupled
+
+  implicit class DecoupledExtensions[A <: Data](x: DecoupledIO[A]) {
```

Comment:
Instead of implementing this as extension methods, can we just implement it as a method directly on `DecoupledIO`?